### PR TITLE
ESQL: Improve tests and docs for some functions

### PR DESCRIPTION
**Pull request by nik9000**
This improves the tests and docs for a few functions, specifically `E`, `FLOOR`, `PI`, `POW`, and `ROUND`. The examples and tested signatures will get copied into the docs and kibana signatures.

### CODE REVIEW

**Review by luigidellaquila — APPROVED** [review]
LGTM

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/Round.java [hunk 1]
```diff
@@ -44,15 +45,26 @@ public class Round extends EsqlScalarFunction implements OptionalArgument {
     private final Expression field, decimals;
 
     // @TODO: add support for "integer", "long", "unsigned_long" once tests are fixed
-    @FunctionInfo(returnType = "double", description = "Rounds a number to the closest number with the specified number of digits.")
+    @FunctionInfo(
+        returnType = { "double", "integer", "long", "unsigned_long" },
+        description = """
+            Rounds a number to the closest number with the specified number of digits.
+            Defaults to 0 digits if no number of digits is provided. If the specified number
+            of digits is negative, rounds to the number of digits left of the decimal point.""",
```
Comment (leemthompo) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/Round.java:53]:
```suggestion
            Rounds a number to the specified number of decimal places.
            Defaults to 0, which returns the nearest integer. If the 
            precision is a negative number, rounds to the number of digits left 
            of the decimal point.""",
```

Comment (leemthompo) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/Round.java:53]: Slightly clearer?

**Review by leemthompo — COMMENTED** [review]
Couple small docs comments and questions, one about markdown syntax :)

File: docs/reference/esql/functions/kibana/docs/e.md [hunk 2]
```diff
@@ -3,5 +3,8 @@ This is generated by ESQL's AbstractFunctionTestCase. Do no edit it. See ../READ
 -->
 
 ### E
-Euler’s number.
+Returns {wikipedia}/E_(mathematical_constant)[Euler's number].
```
[the remark this answers is not in the document]
Comment (nik9000) [reply docs/reference/esql/functions/kibana/docs/e.md:6]: Oh booo. I thought I removed these from the markdown. Will look.

File: docs/reference/esql/functions/kibana/docs/pi.md [hunk 3]
```diff
@@ -3,5 +3,8 @@ This is generated by ESQL's AbstractFunctionTestCase. Do no edit it. See ../READ
 -->
 
 ### PI
-The ratio of a circle’s circumference to its diameter.
+Returns the {wikipedia}/Pi[ratio] of a circle's circumference to its diameter.
```
[the remark this answers is not in the document]
Comment (nik9000) [reply docs/reference/esql/functions/kibana/docs/pi.md:6]: I'm removing the links here - I don't know if they want them in there at all. If they want, we can hack them back in.

**Pull request merged**